### DISCUSSION TITLE
Treat null and empty as same "state"

### DISCUSSION THREAD

**Discussion by ITaluone** [fluentassertions/fluentassertions#1825] [Q&A] [answered]
Hi

I want to test the correct mapping from a DTO to the "real" object.

Now I have following situation: 
A property  (`byte[]`) (which is `null` in the DTO) is mapped as `byte[0]` in the real object. This causes the test to fail, because `null` is not empty. 

As I read, this behavior is expected from AutoMapper. But unfortunatley I can not override the config.

Is it possible somehow to treat this `byte[0]` and `null` as the "same"? (Yes.. I know empty is completely different than null)

```csharp

expected.Should().BeEquivalentTo(actual, options => options
    .ExcludingMissingMembers()
    .WithAutoConversion()
);

```

The fail message:
`Expected property expected.Image to be <null>, but found {empty}.`

**Comment by dennisdoomen** [comment]
Have you tried https://fluentassertions.com/objectgraphs/#equivalency-comparison-behavior?

**Comment by jnyrup** [comment, marked as answer]
Here's an example using the linked documentation, which considers null and empty `byte[]` to be equivalent.

```cs
var subject = new { Value = (byte)null };
var expected = new { Value = new byte[0] };

subject.Should().BeEquivalentTo(expected, opt => opt
    .Using<byte[]>(ctx =>
    {
        if (ctx.Subject is null or { Length: 0 } && ctx.Expectation is null or { Length: 0 })
        {
            return;
        }

        ctx.Subject.Should().BeEquivalentTo(ctx.Expectation);
    })
    .WhenTypeIs<byte[]>()
);
```

**Comment by ITaluone** [comment]
This was very helpful 👍

Thank you both for your great support!

Just realized you could not have multiple answers as accepted answer. Please feel like I have marked both :)

**Reply by rklec** [reply]
Note if you want it to be compared for many types, you may go with a solution like this: https://github.com/fluentassertions/fluentassertions/discussions/1863
